### PR TITLE
MoreUserTags: Fix & Add ability to customize text inside tag

### PR DESCRIPTION
**Pull request by LordEliasTM**
In the attempt of syncing the fork i accidentally pressed discard changes 💀 
Original PR is https://github.com/Vendicated/Vencord/pull/821

### CODE REVIEW

File: src/plugins/moreUserTags.tsx [hunk 1]
```diff
@@ -81,51 +98,89 @@ const tags: Tag[] = [
         permissions: ["MOVE_MEMBERS", "MUTE_MEMBERS", "DEAFEN_MEMBERS"]
     }
 ];
+const defaultSettings = Object.fromEntries(
+    tags.map(({ name, displayName }) => [name, { text: displayName, showInChat: true, showInNotChat: true }])
+) as TagSettings;
+
+function SettingsComponent(props) {
+    if (!settings.store.tagSettings) settings.store.tagSettings = defaultSettings;
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:106]:
```suggestion
    settings.store.tagSettings ??= defaultSettings;
```

File: src/plugins/moreUserTags.tsx [hunk 2]
```diff
@@ -81,51 +98,89 @@ const tags: Tag[] = [
...
+    const [tagSettings, setTagSettings] = useState(settings.store.tagSettings ?? defaultSettings as TagSettings);
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:107]: shouldn't need this due to above line

```suggestion
    const [tagSettings, setTagSettings] = useState(settings.store.tagSettings as TagSettings);
```

File: src/plugins/moreUserTags.tsx [hunk 3]
```diff
@@ -81,51 +98,89 @@ const tags: Tag[] = [
...
+    const setValue = (v: TagSettings) => {
+        setTagSettings(v);
+        props.setValue(v);
+    };
+    return <>
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:112]: multi line components should be wrapped in () and start on a new line, like
```jsx
return (
	<div>
	</div>
)
```

File: src/plugins/moreUserTags.tsx [hunk 4]
```diff
@@ -81,51 +98,89 @@ const tags: Tag[] = [
...
+        {tags.map(t => <Flex flexDirection="column" style={{ gap: 5 }}>
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:113]: missiing a key

File: src/plugins/moreUserTags.tsx [hunk 5]
```diff
@@ -190,6 +245,39 @@ return type!==null?$2.botTag,type"
         },
     ],
 
+    start() {
+        if (!settings.store.tagSettings) {
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:249]: invert this if statement with a return to reduce indent

File: src/plugins/moreUserTags.tsx [hunk 6]
```diff
@@ -190,6 +245,39 @@ return type!==null?$2.botTag,type"
...
+                                newSettings[tag].showInChat = false;
+                                break;
+                            case "never":
+                                newSettings[tag].showInChat = false;
+                                newSettings[tag].showInNotChat = false;
+                                break;
+                        }
+                    }
+                    settings.store.tagSettings = newSettings;
+                    Vencord.Settings.plugins.MoreUserTags[name] = null;
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:274]:
```suggestion
                    delete Vencord.Settings.plugins.MoreUserTags[name];
```

File: src/plugins/moreUserTags.tsx [hunk 7]
```diff
@@ -190,6 +245,39 @@ return type!==null?$2.botTag,type"
...
+                    console.log(Vencord.Settings.plugins.MoreUserTags);
```
Comment (Vendicated) [inline src/plugins/moreUserTags.tsx:275]:
```suggestion
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by ActuallyTheSun** [comment]
threw the fix for canary in here LordEliasTM/Vencord#2 - should work for both canary and stable

**Review by Vendicated — APPROVED** [review]

**Comment by Vendicated** [comment]
Please don't pr from the main branch in the future, it causes so much pain

**Comment by Vendicated** [comment]
Thank you!

**Comment by LordEliasTM** [comment]
> Please don't pr from the main branch in the future, it causes so much pain

What do you mean by that?

**Pull request merged**